### DISCUSSION TITLE
Cast values as text

### DISCUSSION THREAD

**Discussion by FroggyPanda** [supabase/supabase#10731] [Questions] [answered]
How can I cast a value as text?  Currently running into the issue were my UUIDs get cut off at the last two digits and I found [this message](https://discord.com/channels/839993398554656828/869405720934744086/924114790761250826) on the Supabase Discord where the user says "my current workaround is to cast the value to text" but I am confused on how that is actually done.

Table:
![image](https://user-images.githubusercontent.com/101910741/205687346-77887bbd-bc4e-440f-883d-0be274dd676b.png)

Code to get the data:
```js
const { data, error } = await supabase.from('test').select();
```

What is I get back:
```
[{"uuid":340672894898143200,"createdTimestamp":1670216439971},{"uuid":188851079050428400,"createdTimestamp":1670216532008},{"uuid":123456789012345680,"createdTimestamp":3},{"uuid":187469076476787550,"createdTimestamp":3}
```
*The first two rows in the table are with actual Discord UUIDs and timestamps and the last two are random numbers that I put in that had the same length as normal Discord UUIDs

**Comment by GaryAustin1** [comment, marked as answer]
You can cast the value in the .select() like this it turns out.   `select('uuid::text')`   
Then you will get back the entire id as a string and use it.  Javascript can't hold that number so it won't work as a number on javascript side, although I've seen mentions of a library to support bigint.
